src/server/scripts/Kalimdor/zone_silithus.cpp

```diff
@@ -1214,41 +1218,88 @@ class go_wind_stone : public GameObjectScript
+                        if (Group* group = player->GetGroup())
+                        {
+                            ObjectVector secondaryTargets;
+                            for (GroupReference* groupRef = group->GetFirstMember(); groupRef != nullptr; groupRef = groupRef->next())
+                                if (Player* member = groupRef->GetSource())
+                                    if (member != player && member->IsInMap(summons) && member->GetDistance(summons) < 50.0f)
+                                        secondaryTargets.push_back(member);
+
+                            if (secondaryTargets.size() > 0)
+                                sai->GetScript()->AddToStoredTargetList(secondaryTargets, 2);
+                        }
+
+                        switch (summons->GetEntry())
+                        {
                         case NPC_TEMPLAR_FIRE:
                         case NPC_TEMPLAR_WATER:
                         case NPC_TEMPLAR_AIR:
                         case NPC_TEMPLAR_EARTH:
-                            summons->AI()->Talk(SAY_TEMPLAR_AGGRO, player);
+                            sai->Talk(SAY_TEMPLAR_AGGRO, player);
                             break;
 
                         case NPC_DUKE_FIRE:
                         case NPC_DUKE_WATER:
                         case NPC_DUKE_EARTH:
                         case NPC_DUKE_AIR:
-                            summons->AI()->Talk(SAY_DUKE_AGGRO, player);
+                            sai->Talk(SAY_DUKE_AGGRO, player);
                             break;
                         case NPC_ROYAL_FIRE:
                         case NPC_ROYAL_AIR:
                         case NPC_ROYAL_EARTH:
                         case NPC_ROYAL_WATER:
-                            summons->AI()->Talk(YELL_ROYAL_AGGRO, player);
+                            sai->Talk(YELL_ROYAL_AGGRO, player);
                             break;
+                        }
+                        // Next actions happen in smart_scripts
+                        // On Summon: Set Root, Set Immune to PC, Create Timed Event 1 (5s), Create Timed Event 2 (30s)
+                        // On Timed Event 1 (5s): Unset Root, Unset Immune to PC, Add 100.0f Threat (Stored List 1), Add 50.0f Threat (Stored List 2)
+                        // On Reset: Set Counter 1 +1
+                        // On Counter 1 = 2: Despawn Self In 1000ms
+                        // On Timed Event 2 (30s): Despawn Self Instant
+                        // On Aggro: Remove Timed Event 2
+                        // In Combat: Use NPC specific spells
+                    }
+                    else {
+                        TC_LOG_ERROR("scripts", "[zone_silithus] go_wind_stoneAI: NPC `%s` (%s) has wrong AI `%s`, should be `SmartAI`.", summons->GetName().c_str(), summons->GetGUID().ToString().c_str(), summons->GetAIName().c_str());
```

Comment:
Nice to see error logs being added, they help a lot.

If in the future you want to log more information but don't want to write a super long logging function trying to get all the data that might be useful, a lot of classes implement `std::string GetDebugInfo() const` that logs a lot:
- position with x,y,z and map id
- guid, name, entry
- AI
- summoner/charmer
- movement

No matter how spammy this might be with 10/15 lines of data being logged, it gets logged only on errors anyway.

Sometimes I append the GetDebugInfo() result to the end of the log message, so I get the best of both (1st line having the custom log stuff I want to focus on, then the rest).

You don't have to change anything in your code, it's just me sharing information 😃